homeassistant/components/mqtt/device_trigger.py

```diff
@@ -268,14 +267,10 @@ async def async_setup_trigger(
 ) -> None:
     """Set up the MQTT device trigger."""
     config = TRIGGER_DISCOVERY_SCHEMA(config)
-    discovery_hash: tuple[str, str] = discovery_data[ATTR_DISCOVERY_HASH]
-
-    if (device_id := update_device(hass, config_entry, config)) is None:
-        async_dispatcher_send(hass, MQTT_DISCOVERY_DONE.format(discovery_hash), None)
-        return
+    device_id = update_device(hass, config_entry, config)
```

Comment:
Since the schema already validates the data, update device will always return a device_id

Reply:
If this is the case, should the same change be made in `MqttDeviceTrigger.async_update` abovi and `update_device` be retyped?

Reply:
`tag` also uses `update_device` and allows a tag to be set up without device part.

Reply:
> If this is the case, should the same change be made in `MqttDeviceTrigger.async_update` above and `update_device` be retyped?

For an update a schema error is catched from mixins:

https://github.com/home-assistant/core/blob/4142f0d15d39ccfada8effd15e004a108d118cb3/homeassistant/components/mqtt/mixins.py#L722-L725

Reply:
> Since the schema already validates the data, update device will always return a device_id

It will always return a device ID for `device_trigger` since the schema guarantees the `device` config is set correctly.
This makes the code for a `None` as `device_id` unreachable.
For `tag` the `device` config is optional.